src/pagination.rs

```diff
@@ -166,17 +197,30 @@ pub fn generate_paginators(
     match config.include {
         Include::All => {
             sort_posts(&mut all_posts, &config);
-            create_all_paginators(&all_posts, &doc, &config)
+            create_all_paginators(&all_posts, &doc, &config, None)
         }
+        Include::Tags => create_tags_paginators(&all_posts, &doc, &config),
         Include::None => {
             unreachable!("PaginationConfigBuilder should have lead to a None for pagination.")
         }
     }
 }
 
-fn extract_value<'a>(a: &'a liquid::value::Value, key: &str) -> Option<&'a liquid::value::Scalar> {
+fn extract_value<'a>(a: &'a liquid::value::Value, key: &str) -> Option<&'a liquid::value::Value> {
     let key = liquid::value::Scalar::new(key.to_owned());
-    a.get(&key).and_then(|sort_key| sort_key.as_scalar())
+    a.get(&key)
+}
+
+fn extract_scalar<'a>(a: &'a liquid::value::Value, key: &str) -> Option<&'a liquid::value::Scalar> {
+    extract_value(a, key).and_then(|sort_key| sort_key.as_scalar())
+}
+
+fn extract_tags(value: &liquid::value::Value) -> Option<&liquid::value::Array> {
+    if let Some(data) = extract_value(value, "data") {
+        extract_value(data, "tags").and_then(|tags| tags.as_array())
```

Comment:
`data` is meant to be a user-defined.  Part of the intent of this is the concern that we make assumptions on how a field works that the user has created, we could break their site and I'm trying to limit how we break users.

Reply:
I see, I never understood why `categories` was root field and not `tags`, now I do. So this means that this whole index on tags can't really be a thing if we don't know what the user will put in `tags`, can it? :-/

Reply:
We can add `tags` to the frontmatter definition and define the behavior of them (probably its own PR?)

Users would have to migrate from their custom tags to the native tags.

Reply:
Ok, I'll suspend this PR then, and write the RFC for that and the PR after.

Reply:
RFC written: https://github.com/cobalt-org/cobalt.rs/issues/549